src/main/java/com/hubspot/jinjava/objects/serialization/PyishObjectMapper.java

```diff
@@ -40,16 +45,50 @@ public static String getAsUnquotedPyishString(Object val) {
   public static String getAsPyishString(Object val) {
     try {
       return getAsPyishStringOrThrow(val);
-    } catch (JsonProcessingException e) {
+    } catch (IOException e) {
+      if (e instanceof LengthLimitingJsonProcessingException) {
+        if (
+          JinjavaInterpreter
+            .getCurrentMaybe()
+            .map(
+              interpreter -> interpreter.getConfig().getExecutionMode().useEagerParser()
+            )
+            .orElse(false)
+        ) {
+          throw new DeferredValueException(String.format("%s: %s", e.getMessage(), val));
```

Comment:
This is a very good question. It makes me realise I should just add a different method instead. When doing eager execution, if we can't serialize a value properly, then we aren't reconstructing properly for the second render pass. Throwing a DeferredValueException will cause the node in which this is being rendered to become a deferred node.
This will let us know that the pre-render needs to be `REDO`.

But that logic isn't necessary when not using eager execution so I am instead going to make a separate method called `getAsPyishStringOrDefer(Object)` which will wrap the exception in a DeferredValueException

Reply:
Hmm, I think what would be even better is just wrapping the exception in `OutputTooBigException` and then scrapping the pre-render if there are any output too big errors when rendering